consent/manager_test_helpers.go

```diff
@@ -576,21 +576,29 @@ func ManagerTests(m Manager, clientManager client.Manager, fositeManager x.Fosit
 					if i == 5 {
 						c.Client.FrontChannelLogoutURI = "http://some-url.com/"
 					}
-					c.LoginSessionID = "" // otherwise we had to create the login session as well..
-					c.Subject = "subjectLUACWFCL"
+
+					fakeLoginSession := LoginSession{
+						ID:              fmt.Sprintf("fk-login-session-LUACWFCL-%d", i),
+						AuthenticatedAt: time.Now(),
+						Subject:         fmt.Sprintf("subjectLUACWFCL-%d", i),
+					}
+					err = m.CreateLoginSession(context.TODO(), &fakeLoginSession)
+					require.NoError(t, err)
+					c.LoginSessionID = fakeLoginSession.ID // otherwise we had to create the login session as well..
+					c.Subject = fakeLoginSession.Subject
 					clientManager.CreateClient(context.TODO(), c.Client) // Ignore errors that are caused by duplication
 
 					lc, _ := MockAuthRequest(fmt.Sprintf("LUACWFCL-%d", i), true)
 					lc.Challenge = fmt.Sprintf("fk-login-challenge-LUACWFCL-%d", i)
-					lc.SessionID = ""
+					lc.SessionID = fakeLoginSession.ID
 					require.NoError(t, m.CreateLoginRequest(context.TODO(), lc))
 
 					require.NoError(t, m.CreateConsentRequest(context.TODO(), c))
 					_, err = m.HandleConsentRequest(context.TODO(), c.Challenge, h)
 					require.NoError(t, err)
 				}
 
-				clients, err := m.ListUserAuthenticatedClientsWithFrontChannelLogout(context.TODO(), "subjectLUACWFCL")
+				clients, err := m.ListUserAuthenticatedClientsWithFrontChannelLogout(context.TODO(), "subjectLUACWFCL-5", "fk-login-session-LUACWFCL-5")
```

Comment:
The test is misleading in my opinion. So what you're doing is you are creating 5 sessions for 5 different identities and assign that to the same client.

This will always return the one client you expect - unless you call for an invalid session (e.g. `I-AM-NOT-fk-login-session-LUACWFCL-5`), which is why all the tests pass here.

What you should do instead is create two identities with two sessions each, and assign each of these sessions to two clients each. So for example:

- identity 1 + session 1 + client id 1
- identity 1 + session 2 + client id 2
- identity 2 + session 3 + client id 3
- identity 2 + session 4 + client id 4

Now you will be checking against those combinations, so basically:

- Does identity 1 and session 1 give client 1?
- Does identity 1 and session 2 give client 2?
- ...

And you will also need a negative test, so:

- Does identity 1 and session 1 give client 2? -> Expected result is no!
- Does identity 1 and session 2 give client 1? -> Expected result is no!

Reply:
This also applies to the other modified tests!

Reply:
okay